duke/src/main/java/AddCommand.java

```diff
@@ -0,0 +1,149 @@
+import java.text.SimpleDateFormat;
+import java.text.ParseException;
+
+import java.util.Date;
+
+class AddCommand extends Command {
+	private static SimpleDateFormat formatterIn = new SimpleDateFormat("d/M/yyyy HHmm");
+	private static SimpleDateFormat formatterOut = new SimpleDateFormat("d MMM yyyy ha");
+
+	private boolean isLoading = false;
+
+	public AddCommand(String command, String remainingCommand) {
+		super(command, remainingCommand);
+	}
+
+	/**
+	 * Adds new task to list
+	 * @param tasks to add
+	 * @param ui to print
+	 */
+
+	public void execute(TaskList tasks, Ui ui) throws DukeException, ParseException {
+		tasks.addTask(createTask(command, false, remainingCommand));
+
+		ui.print("Got it. I've added this task:\n  " + task.toString());
+		ui.print("Now you have " + tasks.getList().size() + " tasks in the list.");
+	}
+
+	/**
+	 * Caches task associated with command, returns it when queried
+	 * Only called when loading tasks from storage
+	 * @return Task to be added to list using the addCommand
+	 * @throws DukeException  could be thrown by createTask
+	 * @throws ParseException thrown when datetime is unparseable in createTask
+	 */
+
+	public Task getTask() throws DukeException, ParseException {
+		if (this.task == null) {
+			// remainingCommand contains [1/0] [description]
+			isLoading = true;
+			String[] parts = remainingCommand.split(" ");
+			int binary = Integer.valueOf(parts[0]);
+			String description = remainingCommand.substring(2);
+			boolean isDone = (binary == 1);
+			this.task = createTask(command, isDone, description);
+		}
+
+		isLoading = false;
+		return this.task;
+	}
+
+	/**
+	 * Parses command to create task by type with its details, reused by loading
+	 * @param type of task
+	 * @param isDone
+	 * @param description of task, deadline, details, etc
+	 * @return Task created by command
+	 * @throws DukeException  could be thrown by loading invalid tasks
+	 * @throws ParseException could be thrown by unparseable datetime in loading functions
+	 */
+	
+	public Task createTask(String type, boolean isDone, String description) throws DukeException, ParseException {
+		// depending on type
+		switch (type) {
+		case "todo":
+			if (description.isEmpty()) {
+				throw new DukeException("☹ OOPS!!! The description of a " + type + " cannot be empty.");
+			}
+			this.task = loadTodo(isDone, description);
+			break;
+
+		case "deadline":
+			if (description.isEmpty()) {
+				throw new DukeException("☹ OOPS!!! The description of a " + type + " cannot be empty.");
+			}
+			this.task = loadDeadline(isDone, description);
```

Comment:
Why did you choose to do additional parsing inside AddCommand instead of handling it in your Parser class?